tests/mir-opt/dataflow-const-prop/enum.rs

```diff
@@ -68,18 +113,35 @@ fn mutate_discriminant() -> u8 {
 }
 
 // EMIT_MIR enum.multiple.DataflowConstProp.diff
+// CHECK-LABEL: fn multiple(
 fn multiple(x: bool, i: u8) {
+    // CHECK: debug x => [[x:_.*]];
+    // CHECK: debug e => [[e:_.*]];
+    // CHECK: debug x2 => [[x2:_.*]];
+    // CHECK: debug y => [[y:_.*]];
     let e = if x {
+        // CHECK: [[e]] = Option::<u8>::Some(move {{_.*}});
         Some(i)
     } else {
+        // CHECK: [[e]] = Option::<u8>::None;
         None
     };
     // The dataflow state must have:
     //   discriminant(e) => Top
     //   (e as Some).0 => Top
-    let x = match e { Some(i) => i, None => 0 };
-    // Therefore, `x` should be `Top` here, and no replacement shall happen.
-    let y = x;
+    // CHECK-NOT: [[x2]] = const 5
```

Comment:
Why `const 5`?

Reply:
If `x` is not `Top` here, it could potentially be propagated to 5 interprocedually. I can remove if it is confusing.

Reply:
We have no interprocedural const-prop, and we don't plan to. And we check the `const 0`, it can't be both in a single assignment.

Reply:
Sorry for the confusion! Removed.